compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3266,7 +3271,9 @@ public Boolean visit(BType target, BType source) {
                 targetTag = getEffectiveTypeForIntersection(targetType).tag;
             }
             if (isSimpleBasicType(sourceTag) && isSimpleBasicType(targetTag)) {
-                return (source == target) || isIntOrStringType(sourceTag, targetTag);
+                // If type T is ordered, then type T? Is also ordered.
+                return (source == target) || sourceTag == TypeTags.NIL || targetTag == TypeTags.NIL ||
```

Comment:
Do we need nil check here, because we have check `isNil` in `isSameOrderedType` method?

Reply:
But that method will not be called when the both `target`, `source` are simple basic types.